db/migrate/20210103181832_connect_ad_with_city.rb

```diff
@@ -0,0 +1,13 @@
+class ConnectAdWithCity < ActiveRecord::Migration[6.1]
+  def change
+    ActiveRecord::Base.transaction do
+      Rake::Task['db:seed_locations'].invoke
+      rename_column :ads, :city, :city_zdel
+      add_reference :ads, :city, foreign_key: true, index: true
+      Ad.all.each do |ad|
```

Comment:
This loads records in batches, it doesn't necessarily load all ads in the memory. On small heroku instances, and with this many ads now, `Ad.all` might break.

```suggestion
      Ad.find_each do |ad|
```

Reply:
With 79 pages and 20 ads per page I guess there are ~1580 ads and default batch size is 1000, i.e. just two batches. If a small instance might be a problem better do something like `Ad.find_each(batch_size: 100) do ...`